### PR TITLE
scripts to create mdcs cluster

### PR DESCRIPTION
**Pull request by YidingZhou**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
*
*
*
*
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@YidingZhou__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (yidingz). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: matlab-cluster/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,458 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsName": {
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: See #966 for dns guidance

**Comment by singhkay** [comment]
@YidingZhou Are you still working on this?

**Comment by YidingZhou** [comment]
@singhkay This PR is ready now. Thanks.

File: matlab-cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
+Run your MATLAB compute-intensive parallel
```
Comment (singhkay) [inline matlab-cluster/README.md]: Can you add a title for the template readme?

File: matlab-cluster/azuredeploy.json (hunk 1 above)
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: @YidingZhou Can you address this?

File: matlab-cluster/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password. Pick a complex password with uppercase letters, lowercase letters, digits, and symbols. The password should not be longer than 16. Otherwise you'll get a BadRequest error."
+      }
+    },
+    "vmSizeClient": {
+      "type": "string",
+      "metadata": {
+        "description": "Size of the Matlab client nodes."
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Some allowed values would really help the user here

File: matlab-cluster/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+      }
+    },
+    "numWorkersWorker": {
+      "type": "int",
+      "defaultValue": -1,
+      "metadata": {
+        "description": "Number of workers running on each worker VM. -1 will create one worker per core."
+      }
+    },
+    "location": {
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Remove and use resourcegroup().location

File: matlab-cluster/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+        "description": "The URL to the disk image in blob that will be used to create all VMs"
+      }
+    },
+    "scriptUri": {
+      "type": "string",
+      "metadata": {
+        "description": "The URL to VM setup scripts"
+      }
+    },
+    "vhdContainer": {
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Can't this be hardcoded and made a variable?

File: matlab-cluster/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+    "clientScriptName": "[concat(parameters('scriptUri'), 'scripts/private_client_vm_setup.ps1')]",
+    "installationCLI": "powershell -ExecutionPolicy Unrestricted -file private_worker_vm_setup.ps1 shim",
+    "clientInstallationCLI": "powershell -ExecutionPolicy Unrestricted -file private_client_vm_setup.ps1",
+    "networkSecurityGroupName": "NSGMJS",
+    "imageUri": "[parameters('imageUri')]",
+    "vhdContainer": "[parameters('vhdContainer')]"
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Use 2015-06-15 API version

File: matlab-cluster/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+      "name": "[concat(variables('clientVMName'), '/Installation')]",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', variables('clientVMName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.Compute",
+        "type": "CustomScriptExtension",
+        "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Add the autoUpgradeMinorVersion property and use extension v1.5

File: matlab-cluster/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+      "apiVersion": "2015-05-01-preview",
+      "name": "[concat(variables('masterVMName'), '/Installation')]",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', variables('masterVMName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.Compute",
+        "type": "CustomScriptExtension",
+        "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Need to add autoUpgradeMinorVersion and use 1.5 extension version

File: matlab-cluster/azuredeploy.public.json [hunk 9]
```diff
@@ -0,0 +1,378 @@
+{
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.public.json]: What is the purpose of this deploy file?

File: matlab-cluster/azuredeploy.json (hunk 1 above)
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Please also change the parameter name according to #966

File: matlab-cluster/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,596 @@
...
+          "start": 5
+        },
+        "master": "10.0.0.253",
+        "client": "10.0.0.254"
+      }
+    },
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('networkSettings').virtualNetworkName)]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('networkSettings').subnet.dse.name)]",
+    "sshKeyPath": "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+    "scriptUri": "https://raw.githubusercontent.com/YidingZhou/azure-quickstart-templates/master/matlab-cluster/",
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: Script needs to point to Azure (this repo) location

File: matlab-cluster/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,596 @@
...
+      "name": "[concat(variables('clientVMName'), '/Installation')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', variables('clientVMName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.Compute",
+        "type": "CustomScriptExtension",
+        "typeHandlerVersion": "1.5",
+        "autoUpgradeMinorVersion": "true",
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: this is a boolean property

File: matlab-cluster/azuredeploy.json (hunk 7 above)
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: This needs to be addressed

File: matlab-cluster/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,596 @@
...
+      ],
+      "copy": {
+        "name": "foo",
+        "count": "[parameters('numWorkerVms')]"
+      },
+      "properties": {
+        "publisher": "Microsoft.Compute",
+        "type": "CustomScriptExtension",
+        "typeHandlerVersion": "1.5",
+        "autoUpgradeMinorVersion": "true",
```
Comment (singhkay) [inline matlab-cluster/azuredeploy.json]: boolean

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**